pennylane/workflow/execution.py

```diff
@@ -0,0 +1,51 @@
+# Copyright 2018-2023 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Contains the :class:`ExecutionConfig` data class.
+"""
+from dataclasses import dataclass, field
+from typing import Callable, Union
+
+from pennylane.interfaces import SUPPORTED_INTERFACES
+
+
+@dataclass
+class ExecutionConfig:
+    """
+    A configuration class to describe an execution of a quantum circuit on a device.
+    """
+
+    shots: Union[None, int] = None
```

Comment:
I learned recently about `typing.Optional`. Instead of `Union[None, int]`, we can use `Optional[int]`.

Additional question, should we store shot vectors in the `shots` property or in an additional `shot_vector` property? If so, shots could also be `Tuple[int]`.

Reply:
I originally had shots as a union with tuple, but I didn't really see any previous examples for `shots` like this. I prefer it though, so I'll put it back. `Optional[Union[int, Tuple[int]]]` lol